### PR TITLE
Added support for the Arduino Primo Core

### PR DESCRIPTION
**Pull request by jeremypoulter**
Added support for the Arduino Primo Core. Basically grabbed Arduino.org's pin map from https://github.com/arduino-org/arduino-core-nrf52.

### CODE REVIEW

**Comment by dlabun** [comment]
Before this can be merged the boards.txt and Readme also need to be updated. If you can update them that would be great, otherwise this will have to wait until somebody else is able to.

**Comment by jeremypoulter** [comment]
Sure can do this, however with regard to the boards.txt, I added this primarily to get Primo Core support in Platform IO. For use with Arduino IDE would you not want to use Arduino.orgs core in preference? Adding to boards.txt could add confusion?

File: variants/PrimoCore/variant.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+/*
+ * Pins descriptions
+ */
+//const PinDescription g_APinDescription[]=
+const uint32_t g_ADigitalPinMap[] = 
+{
+  // 0 .. 9 - Digital pins
+  // ----------------------
+  // 0..9
+  31, // { PORT0,  31, PIO_DIGITAL, (PIN_ATTR_DIGITAL|PIN_ATTR_PWM), ADC_A5, PWM0, NOT_ON_TIMER},
```
Comment (sandeepmistry) [inline variants/PrimoCore/variant.cpp]: Can we remove the unneeded comments in this file?

File: variants/PrimoCore/variant.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  18, // { PORT0, 18, PIO_DIGITAL, PIN_ATTR_DIGITAL, No_ADC_Channel, NOT_ON_PWM, NOT_ON_TIMER}, // TP5
+  
+  //27..28 - NFC
+  // --------------------
+  9, // { PORT0, 9, PIO_DIGITAL, PIN_ATTR_DIGITAL, No_ADC_Channel, NOT_ON_PWM, NOT_ON_TIMER}, // NFC1
+  10, // { PORT0, 10, PIO_DIGITAL, PIN_ATTR_DIGITAL, No_ADC_Channel, NOT_ON_PWM, NOT_ON_TIMER} // NFC2
+
+} ;
+
+//Uart Serial(14, 15);
```
Comment (sandeepmistry) [inline variants/PrimoCore/variant.cpp]: Why is the UART not declared?

File: variants/PrimoCore/variant.h [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+/*----------------------------------------------------------------------------
+ *        Pins
+ *----------------------------------------------------------------------------*/
+
+// Number of pins defined in PinDescription array
+#define PINS_COUNT           (16u)
+#define NUM_DIGITAL_PINS     (8u)
+
+//#define digitalPinToPort(P)        ( NRF_P0 )
```
Comment (sandeepmistry) [inline variants/PrimoCore/variant.h]: Do these comments need to be here?

**Comment by sandeepmistry** [comment]
> Sure can do this, however with regard to the boards.txt, I added this primarily to get Primo Core support in Platform IO. 

I'm not familiar with Platform IO, how can the variant folder files be used without a boards.txt entry?

> For use with Arduino IDE would you not want to use Arduino.orgs core in preference? Adding to boards.txt could add confusion?

I'm confused, why is this PR is open if this is a concern then?

File: boards.txt [hunk 4]
```diff
@@ -252,6 +252,37 @@ STCT_nRF52_minidev.menu.softdevice.s132.build.extra_flags=-DNRF52 -DS132 -DNRF51
...
 
 
+Arduino_Primo_Core.name=Arduino Primo Core
+
+Arduino_Primo_Core.upload.tool=sandeepmistry:openocd
+Arduino_Primo_Core.upload.target=nrf52
+Arduino_Primo_Core.upload.maximum_size=524288
+
+Arduino_Primo_Core.bootloader.tool=sandeepmistry:openocd
+
```
Comment (sandeepmistry) [inline boards.txt]: Does the Primo Core have a built-in programmer? (CMSIS DAP, JLINK, ST-LINK, etc?)

if so, a line should be added:
```
Arduino_Primo_Core.upload.protocol=<blah>
```

File: variants/PrimoCore/variant.cpp (hunk 1 above)
Comment (jeremypoulter) [reply variants/PrimoCore/variant.cpp]: Sure, just a hangover from Arduino's variant. Will remove all the commented out code.

File: variants/PrimoCore/variant.h (hunk 3 above)
Comment (jeremypoulter) [reply variants/PrimoCore/variant.h]: as above

File: variants/PrimoCore/variant.cpp (hunk 2 above)
Comment (jeremypoulter) [reply variants/PrimoCore/variant.cpp]: Not needed for this Core, Arduino's is structured slightly different

File: boards.txt (hunk 4 above)
Comment (jeremypoulter) [reply boards.txt]: The Primo Core needs an external programmer

**Comment by jeremypoulter** [comment]
Platform IO uses its own version of the build.txt file, so not needed. The reason I added this was two fold, it is not immediately obvious (in Platform IO) how to use multiple Arduino cores for the same chipset framework and they have already integrated this core for the Arduino/nRF52, and second is the Primo Core was a small stepping stone in our product development, both the prototype and subsequent versions will be based on generic nRF52 and use this core so I wanted to use this for the Primo Core version as well.

**Comment by dlabun** [comment]
@sandeepmistry this one is bugging me... Do we really want to start supporting Arduino brand boards?

**Comment by sandeepmistry** [comment]
@dlabun good point, let's leave this open for now and see what the demand is for it.

**Comment by ttn115** [comment]
Hi, I have try the guide of README.md, but I was not able to flash my arduino primo core, 
anymore detail or advice?
https://github.com/sandeepmistry/arduino-nRF5/issues/217

**Comment by dlabun** [comment]
Closing due to apparent lack of interest in supporting the Primo.

**Pull request closed without merging**